src/components/map/MapService.js

```diff
@@ -827,6 +828,10 @@ goog.require('ga_urlutils_service');
           }
         };
 
+        this.getLayers = function() {
+          return layersPromise;
+        };
+
```

Comment:
Maybe this promise could be useful in gaLayersPermalinkManger too. Using $q.all that simplify a lot what happens on first load.